kafka/conn.py

```diff
@@ -357,6 +357,10 @@ def connect(self):
                 ret = self._sock.connect_ex(self._sock_addr)
             except socket.error as err:
                 ret = err.errno
+            except ValueError:
```

Comment:
Given that this is python 3.7 and above, and that `ValueError` is a common enough error, I'd like to minimize the chance of this swallowing something unexpected... 

So could you also add a check that we are running under python >= 3.7? And if not, re-raise and don't set `ret`

See here for details: https://stackoverflow.com/a/1093331/770425